core/arch/arm/plat-nuvoton/main.c

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2016-2023, Linaro Limited
+ * Copyright (c) 2014-2023, STMicroelectronics International N.V.
+ * Copyright (C) 2022-2023 Nuvoton Ltd.
+ */
+
+#include <console.h>
+#include <drivers/gic.h>
+#include <drivers/ns16550.h>
+#include <kernel/boot.h>
+#include <kernel/interrupt.h>
+#include <kernel/linker.h>
+#include <mm/core_mmu.h>
+#include <platform_config.h>
+#include <trace.h>
+
+static struct gic_data gic_data __nex_bss;
+static struct ns16550_data console_data __nex_bss;
+
+register_phys_mem_pgdir(MEM_AREA_IO_SEC, CONSOLE_UART_BASE, UART_REG_SIZE);
+register_phys_mem_pgdir(MEM_AREA_IO_SEC, GICD_BASE, GIC_DIST_REG_SIZE);
+register_phys_mem_pgdir(MEM_AREA_IO_SEC, GICC_BASE, GIC_DIST_REG_SIZE);
+
+register_ddr(DRAM0_BASE, DRAM0_SIZE);
+
+static void print_version(void)
+{
+	MSG(COLOR_MAGENTA);
```

Comment:
There are special tools available to colorize a log, I don't see why it needs to be hardcoded into firmware.

Reply:
because it's not log, it's an debug output in connected console

Reply:
That's a shortcoming in the tools you're using.